packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -136,6 +139,19 @@ export class RoutesMatcher {
 			return;
 		}
 
+		// If the new destination is an intercept route, only allow it if the current path is also an
+		// intercept route. The build output config correctly maps relevant request paths to be
+		// intercepts in the `none` phase, while the `rewrite` phase can contain entries that rewrite
+		// to an intercept that matches requests that are not actually intercepts, causing a 404.
+		if (
+			checkIntercept &&
+			route.dest &&
+			/\/(\(\.+\))+/.test(route.dest) &&
+			!/\/(\(\.+\))+/.test(this.path)
```

Comment:
I don't thing I understand the comment above in regards of these, the comment says
"if the new destination is an intercept route, only allow it if the current path is also an intercept route"
but here is looks to me like the dest is an intercept route and the current path is not, am I missing something?

Reply:
yeah so what im doing here is not letting the path be rewritten to the intercept dest if the path isn't already an intercept

so i guess the comment is kind of the inverse?

Reply:
happy to change it something clearer

Reply:
ah ok I see! sorry my bad I got confused by it...

So I think that the comment is ok indeed, but what about maybe changing this to something like this:
![Screenshot 2023-11-29 at 09 25 01](https://github.com/cloudflare/next-on-pages/assets/61631103/a4bac634-f237-410b-a38f-4d15aae85722)

In this way the `if` check becomes very clear and the comment also follows the same structure all the other ones in this function do

what do you think?

Reply:
looks like a good plan to me